### PR TITLE
Allow multiple users on CalLogSummary

### PR DESCRIPTION
**Pull request by Shanefe**
**Issue:**
Increase efficiency by passing in multiple users (Organizer, Delegate, attendee, Room, etc.)

**Reason:**
I always need all the CalLogs, might as well collect them all at once. 

**Fix:**
Loop over the users, supported only with MeetingID.

**Validation:**
Tested with and without PII, Torus and with Tenant PS.

### CODE REVIEW

**Comment by dpaulson45** [comment]
@Shanefe is this ready for review or you still making changes to it?

**Comment by Shanefe** [comment]
Should be good to go. I just had a case where I found a hole in the coverage.

Shane

Is there a proper places to store validation scripts?

**Comment by Shanefe** [comment]
OK, should be good to go now that I have removed the trailing spaces.  Those are going to kill me one day...

From: Shane Ferrell
Sent: Thursday, November 16, 2023 11:22 AM
To: microsoft/CSS-Exchange ***@***.***>; microsoft/CSS-Exchange ***@***.***>
Cc: Mention ***@***.***>
Subject: RE: [microsoft/CSS-Exchange] Allow multiple users on CalLogSummary (PR #1877)

Should be good to go. I just had a case where I found a hole in the coverage.

Shane

Is there a proper places to store validation scripts?

File: Calendar/Get-CalendarDiagnosticObjectsSummary.ps1 [hunk 1]
```diff
@@ -1234,67 +1255,155 @@ function BuildTimeline {
...
 }
 
-Write-Host -ForegroundColor Yellow "`n`n`n============================================================================"
-Write-Host -ForegroundColor Yellow "Hope this script was helpful in getting (and understanding) the Calendar Logs."
-Write-Host -ForegroundColor Yellow "If you have issues or suggestion for this script,"
-Write-Host -ForegroundColor Yellow "`t please send them to <callogformatterdevs@microsoft.com>"
-Write-Host -ForegroundColor Yellow "============================================================================`n`n`n"
+Write-DashLineBoxColor  "Hope this script was helpful in getting and understanding the Calendar Logs.",
+"If you have issues or suggestion for this script, please send them to: ",
+"`t <Shanefe@Microsoft.com" -Color Yellow -DashChar =
```
Comment (dpaulson45) [inline Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:1409]: Recommend changing this to a group of some kind. Helps protects your email and in case you leave, the script doesn't need to be updated.

File: Calendar/Get-CalendarDiagnosticObjectsSummary.ps1 [hunk 2]
```diff
@@ -138,28 +139,30 @@ $ResponseTypeOptions = @{
...
+    param(
+        [string]$Identity,
+        [string]$Subject,
+        [string]$MeetingID
+    )
     # Use MeetingID if we have it.
     if ($Identity -and $MeetingID) {
         Write-Verbose "Getting CalLogs for [$Identity] with MeetingID [$MeetingID]."
-        $script:InitialCDOs = Get-CalendarDiagnosticObjects -Identity $Identity -MeetingID $MeetingID -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults $LogLimit -ResultSize $LogLimit -ShouldBindToItem $true;
+        $CalLogs = Get-CalendarDiagnosticObjects -Identity $Identity -MeetingID $MeetingID -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults $LogLimit -ResultSize $LogLimit -ShouldBindToItem $true;
```
[the remark this answers is not in the document]
Comment (Shanefe) [reply Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:150]: Done.

File: Calendar/Get-CalendarDiagnosticObjectsSummary.ps1 [hunk 3]
```diff
@@ -138,28 +139,30 @@ $ResponseTypeOptions = @{
...
     }
 
     # Otherwise do a search on the subject.
-    if ($Identity -and $Subject -and !$MeetingID) {
+    if ($Identity -and $Subject) {
```
[the remark this answers is not in the document]
Comment (Shanefe) [reply Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:154]: When using a parameter set, it validates that both Subject and MeetingID are not passed in.
.\Get-CalendarDiagnosticObjectsSummary.ps1 Delegate -subject "Lunch" -MeetingID $MeetingID
_Get-CalendarDiagnosticObjectsSummary.ps1: Parameter set cannot be resolved using the specified named parameters. One or more parameters issued cannot be used together or an insufficient number of parameters were provided._

File: Calendar/Get-CalendarDiagnosticObjectsSummary.ps1 (hunk 1 above)
Comment (Shanefe) [reply Calendar/Get-CalendarDiagnosticObjectsSummary.ps1]: Done.  I had to enable external addresses.

File: Calendar/Get-CalendarDiagnosticObjectsSummary.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply Calendar/Get-CalendarDiagnosticObjectsSummary.ps1]: Parameter sets is different than what I was talking about here. But you are able to use Parameter Sets as well. Because `Delegate` wasn't provided a parameter name, you might need to provide a position for the parameters to know what goes where.

File: Calendar/Get-CalendarDiagnosticObjectsSummary.ps1 [hunk 4]
```diff
@@ -754,15 +770,23 @@ function BuildCSV {
...
+        $ShortName = $Identity.Split('@')[0]
+    }
+    $ShortName = $ShortName.Substring(0, [System.Math]::Min(20, $ShortName.Length))
+    $Filename = "$($ShortName)_$ShortMeetingID.csv";
+    Write-Host -ForegroundColor Cyan -NoNewline "Calendar Logs for [$Identity] have been saved to :"
+    Write-Host -ForegroundColor Yellow "$Filename"
     $GCDOResults | Export-Csv -Path $Filename -NoTypeInformation -Encoding UTF8
 
-    $MeetingTimeLine = $Results | Where-Object { $_.IsIgnorable -eq "False" } ;
+    $MeetingTimeLine = $Results | Where-Object { $Results.IsIgnorable -eq "False" } ;
```
Comment (dpaulson45) [inline Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:783]: This `Where-Object` cause is incorrect now. Why was this changed?

**Review by dpaulson45 — APPROVED** [review]
